libtextsecure/outgoing_message.js

```diff
@@ -362,20 +389,46 @@ OutgoingMessage.prototype = {
           sourceDevice: 1,
           destinationRegistrationId: ciphertext.registrationId,
           content: ciphertext.body,
+          number: devicePubKey,
         };
       })
     )
       .then(async outgoingObjects => {
         // TODO: handle multiple devices/messages per transmit
-        const outgoingObject = outgoingObjects[0];
-        const socketMessage = await this.wrapInWebsocketMessage(outgoingObject);
-        await this.transmitMessage(
-          number,
-          socketMessage,
-          this.timestamp,
-          outgoingObject.ttl
-        );
-        this.successfulNumbers[this.successfulNumbers.length] = number;
+        let counter = 0;
+        const promises = outgoingObjects.map(async outgoingObject => {
+          const destination = outgoingObject.number;
+          try {
+            counter += 1;
+            if (counter > 1) {
```

Comment:
If I'm understanding this logic it is just trying to ensure that we only process the first outgoing object?
Seems strange, if this error gets thrown then the promise all will fail right? Not sure I get it

Reply:
OOps this was test code that I failed to clean up!